numcodecs/zarr3.py

```diff
@@ -0,0 +1,282 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, replace
+from functools import cached_property
+import math
+from typing_extensions import Self
+from warnings import warn
+
+import numpy as np
+import numcodecs
+
+try:
+    import zarr
+
+    assert zarr.__version__ >= "3.0.0"
```

Comment:
There should be a test that this errors as expected (a nice import error) if zarr v2 is installed.